packages/core/src/core/coreToolScheduler.ts

```diff
@@ -893,8 +896,32 @@ export class CoreToolScheduler {
             : undefined;
 
         const shellExecutionConfig = this.config.getShellExecutionConfig();
-        invocation
-          .execute(signal, liveOutputCallback, shellExecutionConfig)
+
+        let promise: Promise<ToolResult>;
+        if (invocation instanceof ShellToolInvocation) {
+          const setPidCallback = (pid: number) => {
```

Comment:
special casing this for ShellToolInvocation is a bit ugly. would be ideal if there is a simpler way but won't block the pr on that. perhaps add a todo.